sdk/src/client/node_api/core/routes.rs

```diff
@@ -310,109 +299,101 @@ impl ClientInner {
-    ) -> Result<UtxoChangesFullResponse> {
-        let path = &format!("api/core/v3/commitments/{slot_commitment_id}/utxo-changes/full");
+    pub async fn get_utxo_changes_full(&self, commitment_id: &SlotCommitmentId) -> Result<UtxoChangesFullResponse> {
+        let path = &format!("api/core/v3/commitments/{commitment_id}/utxo-changes/full");
 
         self.get_request(path, None, false).await
     }
 
     /// Finds a slot commitment by slot index and returns it as object.
     /// GET /api/core/v3/commitments/by-slot/{slot}
-    pub async fn get_slot_commitment_by_slot(&self, slot_index: SlotIndex) -> Result<SlotCommitment> {
-        let path = &format!("api/core/v3/commitments/by-slot/{slot_index}");
+    pub async fn get_commitment_by_slot(&self, slot: SlotIndex) -> Result<SlotCommitment> {
+        let path = &format!("api/core/v3/commitments/by-slot/{slot}");
 
         self.get_request(path, None, false).await
     }
 
     /// Finds a slot commitment by slot index and returns it as raw bytes.
     /// GET /api/core/v3/commitments/by-slot/{slot}
-    pub async fn get_slot_commitment_by_slot_raw(&self, slot_index: SlotIndex) -> Result<Vec<u8>> {
-        let path = &format!("api/core/v3/commitments/by-slot/{slot_index}");
+    pub async fn get_commitment_by_slot_raw(&self, slot: SlotIndex) -> Result<Vec<u8>> {
+        let path = &format!("api/core/v3/commitments/by-slot/{slot}");
 
         self.get_request_bytes(path, None).await
     }
 
     /// Get all UTXO changes of a given slot by its index.
     /// GET /api/core/v3/commitments/by-slot/{slot}/utxo-changes
-    pub async fn get_utxo_changes_by_slot(&self, slot_index: SlotIndex) -> Result<UtxoChangesResponse> {
-        let path = &format!("api/core/v3/commitments/by-slot/{slot_index}/utxo-changes");
+    pub async fn get_utxo_changes_by_slot(&self, slot: SlotIndex) -> Result<UtxoChangesResponse> {
+        let path = &format!("api/core/v3/commitments/by-slot/{slot}/utxo-changes");
 
         self.get_request(path, None, false).await
     }
 
     /// Get all full UTXO changes of a given slot by its index.
     /// GET /api/core/v3/commitments/by-slot/{slot}/utxo-changes/full
-    pub async fn get_utxo_changes_full_by_slot(&self, slot_index: SlotIndex) -> Result<UtxoChangesFullResponse> {
-        let path = &format!("api/core/v3/commitments/by-slot/{slot_index}/utxo-changes/full");
+    pub async fn get_utxo_changes_full_by_slot(&self, slot: SlotIndex) -> Result<UtxoChangesFullResponse> {
+        let path = &format!("api/core/v3/commitments/by-slot/{slot}/utxo-changes/full");
 
         self.get_request(path, None, false).await
     }
 }
 
 impl Client {
     /// GET /api/core/v3/info endpoint
-    pub(crate) async fn get_permanode_info(mut node: Node) -> Result<PermanodeInfoResponse> {
-        log::debug!("get_permanode_info");
-        if let Some(auth) = &node.auth {
+    pub async fn get_info(url: &str, auth: Option<NodeAuth>) -> Result<InfoResponse> {
```

Comment:
hmm...maybe instead just accept a `Url` type?

Reply:
I would say yes, but not sure this change should be part of this PR.

Reply:
fair enough I suppose